### ISSUE TITLE
Timber image resize problem

### ISSUE THREAD

**Issue by zcuric** [timber/timber#1706]
Hi, how does image resize actually work? 
How Timber knows which image to pull when image is resized? Is this information stored in the database? If so, where? From code it seems a bit unclear. 

Also, is it possible to make re-resize - delete the previoue resized image and replace it with a new one?
If not, is possible to make a function that does that? I would personally try to make plugin for that.

I'm asking this because, I use resize filter instead of `add_image_size` and sometimes resize image fails, not often and I'm not able to resize it again, the old broken resized image is always there.

**Comment by gchtr** [comment]
Hey @zcuric 

In https://github.com/timber/timber/issues/754 the same topic was brought up.

To your question about how resizing works: Timber will generate file names based on the arguments you pass to the resize filter. When an image with a certain name exists already, it will return that image, otherwise, it will generate the size on the fly.

When you delete an attachment in the backend, Timber will check if resized versions of the image exist and delete those files through [Timber\ImageHelper::delete_generated_files](https://github.com/timber/timber/blob/8d298aaa88cbab56b44f14968c5da781172f8e84/lib/ImageHelper.php#L227-L244).

You can also use this function directly, if you pass the original file source. But there’s not an easy way to resize an image with the click of a button yet. I think it would be a nice addition if there was a plugin that adds a function to delete generated files for an image.  
The [Regenerate Thumbnails](https://wordpress.org/plugins/regenerate-thumbnails/) plugin would offer the functionality, but Timber doesn’t integrate with it. You could look how it integrates into the backend, or you could write an integration to make it work with Timber by using the `wp_generate_attachment_metadata` filter. However, it will only be possible to delete generated sizes, but not to generate them again. You’d have to reload the page where the images are used to do that.

**Comment by zcuric** [comment]
Thanks for the feedback @gchtr!

Let me give you my feedback on so several issues I encountered while playing with images. 

So my site uses Jetpack Photon with Timber. Because of the resizing of images, one doesn't not actually benefit from TimberPhoton plugin, to save bandwidth because the dynamic resizing (more info here: [PHOTON: WHERE IS MY BANDWIDTH GOING?](https://www.chromosphere.co.uk/2015/11/07/photon-where-is-my-bandwidth-going/).

So what I did, is I used the native Timber resizing to resize the images on my server and let Photon serve them. Which actually worked and saved the bandwidth. 

During some update of one of the plugins, Jetpack or Timber, in December, this stopped working. My guess is that `post.thumbnail.src` started returning the Photon image url instead of the server url, and the resizing did not work as expected and the bandwidth started going back up again. See the image. 

![image](https://user-images.githubusercontent.com/1560102/38770842-4e047374-4019-11e8-9d37-1d8dbcf7411d.png)

Now, to reverse the behavior, I had to use `post.thumbail.guid`, resize the image and set up the Photon url manually. For now it works. 

To re-resize the images, I made a test script that accepts `year`, `month` arguments to delete the all images which are re-resized again on page load. Main part of the script is: 
```
function deleteTimberResizedImages($year, $month) {
  $args = [
    'post_type'      => 'attachment',
    'post_mime_type' => 'image',
    'post_status'    => 'inherit',
    'posts_per_page' => - 1,
    'year' => $year,
    'monthnum' => $month
  ];

  $queryImages = new WP_Query($args);
  $images = [];

  foreach ($queryImages->posts as $image) {
    $images[] = wp_get_attachment_url($image->ID);
  }

  foreach ($images as $image) {
    Timber\ImageHelper::delete_generated_files($image);
  }
}
```

This can be improve obviously, but for now it did the job. I'm planing to make this into a simple plugin that will be called from settings and on post individually. 

Also, I had some issue on php 7.1., no image editor was available to resize the images. Could be related with my server. 

I'll update this issue when I make the plugin.
